app/questionnaire/rules.py

```diff
@@ -170,14 +170,16 @@ def evaluate_when_rules(when_rules, metadata, answer_store, group_instance):
     :param metadata: metadata for evaluating rules with metadata conditions
     :param answer_store: store of answers to evaluate
     :param group_instance: when evaluating a when rule for a repeating group
+           this holds the group instance of the repeating group
     :return: True if the when condition has been met otherwise False
     """
     for when_rule in when_rules:
         if 'id' in when_rule:
             when_specified_group_instance = when_rule.get('group_instance')
```

Comment:
This could be:

```
target_group_instance = when_rule.get('group_instance', group_instance)
```

(renamed to `target_group_instance` to be explicit)

or

```
if 'group_instance' in when_rule:
    group_instance = when_rule['group_instance']
```

The latter may be clearer. I don't think we should need to explictly check for `None` as the only way this variable could hold that value is if group_instance wasn't defined for the when_rule.